### PR TITLE
Avoid usage of exports/module.exports/require().

### PR DESCRIPTION
**Pull request by loganfsmyth**
Just trying to standardize on ES6 module syntax. Now the only places we use `require()` are truly dynamic `require()` calls for plugin loading, and a few places where we have optional dependencies that may fail to load.

### CODE REVIEW

**Comment by loganfsmyth** [comment]
@danez Yeah, cycles were definitely a concern, but it seems like that didn't end up being an issue for our usecases.

**Comment by danez** [comment]
I used madge to detect circular dependencies and found some, but nearly all are probably okay. They are all within their own package, because madge doesn't understand the multirepo.

```
✖ Found 29 circular dependencies!

1) babel-cli/src/babel/dir > babel-cli/src/babel/util > babel-cli/src/babel/index
2) babel-cli/src/babel/util > babel-cli/src/babel/index > babel-cli/src/babel/file
3) babel-core/src/transformation/file/index > babel-core/src/transformation/file/logger
4) babel-core/src/index > babel-core/src/transformation/file/index > babel-core/src/transformation/file/options/option-manager
5) babel-core/src/transformation/file/options/option-manager > babel-core/src/transformation/plugin
6) babel-core/src/transformation/file/index > babel-core/src/transformation/pipeline
7) babel-core/src/transformation/file/index > babel-core/src/transformation/plugin-pass
8) babel-traverse/src/context > babel-traverse/src/path/index
9) babel-traverse/src/context > babel-traverse/src/path/index > babel-traverse/src/index
10) babel-traverse/src/path/index > babel-traverse/src/index
11) babel-traverse/src/index > babel-traverse/src/scope/index
12) babel-traverse/src/path/index > babel-traverse/src/index > babel-traverse/src/scope/index
13) babel-traverse/src/path/index > babel-traverse/src/index > babel-traverse/src/scope/index > babel-traverse/src/scope/binding
14) babel-traverse/src/path/index > babel-traverse/src/index > babel-traverse/src/visitors > babel-traverse/src/path/lib/virtual-types
15) babel-traverse/src/path/index > babel-traverse/src/path/ancestry
16) babel-traverse/src/path/index > babel-traverse/src/path/evaluation
17) babel-traverse/src/path/index > babel-traverse/src/path/family
18) babel-traverse/src/path/inference/index
19) babel-traverse/src/path/index > babel-traverse/src/path/inference/index > babel-traverse/src/path/inference/inferers > babel-traverse/src/path/inference/inferer-reference
20) babel-traverse/src/path/index > babel-traverse/src/path/introspection
21) babel-traverse/src/path/index > babel-traverse/src/path/modification
22) babel-traverse/src/path/index > babel-traverse/src/path/replacement
23) babel-types/src/converters > babel-types/src/index
24) babel-types/src/index > babel-types/src/definitions/index
25) babel-types/src/index > babel-types/src/definitions/init > babel-types/src/definitions/core
26) babel-types/src/index > babel-types/src/flow
27) babel-types/src/index > babel-types/src/react
28) babel-types/src/index > babel-types/src/retrievers
29) babel-types/src/index > babel-types/src/validators
```

File: packages/babel-traverse/src/path/index.js [hunk 1]
```diff
@@ -9,6 +9,19 @@ import Scope from "../scope";
 import * as t from "babel-types";
 import { path as pathCache } from "../cache";
 
+// NodePath is split across manu files.
```
Comment (Kovensky) [inline packages/babel-traverse/src/path/index.js]: typo

File: packages/babel-plugin-transform-regenerator/src/index.js [hunk 2]
```diff
@@ -1 +1,2 @@
-module.exports = require("regenerator-transform");
+import transform from "regenerator-transform";
+export default transform;
```
Comment (Kovensky) [inline packages/babel-plugin-transform-regenerator/src/index.js]: `export { default } from "regenerator-transform";`

**Comment by zertosh** [comment]
Ah cool, then we can use https://github.com/zertosh/babel-plugin-transform-inline-imports-commonjs

**Comment by Kovensky** [comment]
That transform is actually kinda cool. Have to be careful, though, as that is only correct if all modules' bodies are pure.

**Comment by zertosh** [comment]
@Kovensky, true - but in practice, it hasn't really been an issue. We've been using it in https://github.com/facebook/nuclide for ~8 months, and in yarn since before it was open sourced.

**Comment by hzoo** [comment]
aside: wish it was easier to track the plugins people are using

**Pull request merged**